src/main/java/io/vertx/ext/mail/MailConfig.java

```diff
@@ -151,17 +145,33 @@ public MailConfig(JsonObject config) {
 
     username = config.getString("username");
     password = config.getString("password");
-    ssl = config.getBoolean("ssl", DEFAULT_SSL);
-    trustAll = config.getBoolean("trustAll", DEFAULT_TRUST_ALL);
-    keyStore = config.getString("keyStore");
-    keyStorePassword = config.getString("keyStorePassword");
+    // Handle these for compatiblity
+    if (config.containsKey("keyStore")) {
+      setKeyStore(config.getString("keyStore"));
+    }
+    if (config.containsKey("keyStorePassword")) {
+      setKeyStorePassword(config.getString("keyStorePassword"));
+    }
     authMethods = config.getString("authMethods");
     ownHostname = config.getString("ownHostname");
     maxPoolSize = config.getInteger("maxPoolSize", DEFAULT_MAX_POOL_SIZE);
     keepAlive = config.getBoolean("keepAlive", DEFAULT_KEEP_ALIVE);
     allowRcptErrors = config.getBoolean("allowRcptErrors", DEFAULT_ALLOW_RCPT_ERRORS);
   }
 
+  @Override
```

Comment:
This maintains compatibility with the way the underlying NetClientOptions were created in the SMTPConnectionPool code (while still allowing someone to override this behavior if they choose to).